Intersect.Server/Entities/Npc.cs

```diff
@@ -735,394 +712,420 @@ public override void Update(long timeMs)
             try
             {
                 Monitor.TryEnter(EntityLock, ref lockObtained);
-                if (lockObtained)
+                if (!lockObtained)
                 {
-                    var curMapLink = MapId;
-                    base.Update(timeMs);
-                    var tempTarget = Target;
+                    return;
+                }
 
-                    foreach (var status in CachedStatuses)
+                var curMapLink = MapId;
+                base.Update(timeMs);
+                var tempTarget = Target;
+
+                foreach (var status in CachedStatuses)
+                {
+                    if (status.Type == SpellEffect.Stun || status.Type == SpellEffect.Sleep)
                     {
-                        if (status.Type == SpellEffect.Stun || status.Type == SpellEffect.Sleep)
-                        {
-                            return;
-                        }
+                        return;
                     }
+                }
 
-                    var fleeing = IsFleeing();
+                var fleeing = IsFleeing();
 
-                    if (MoveTimer < Timing.Global.Milliseconds)
+                if (MoveTimer < Timing.Global.Milliseconds)
+                {
+                    var targetMap = Guid.Empty;
+                    var targetX = 0;
+                    var targetY = 0;
+                    var targetZ = 0;
+
+                    //TODO Clear Damage Map if out of combat (target is null and combat timer is to the point that regen has started)
+                    if (tempTarget != null && (Options.Instance.NpcOpts.ResetIfCombatTimerExceeded &&
```

Comment:
hmm.. the original check only evaluate the second condition if tempTarget is not null, while the suggested change will evaluate all three conditions regardless of the value of tempTarget, this would mean a different behavior from what's originally here and more checks at the same time than just the first one(?), maybe we shouldn't touch this one  😟 ❓

Reply:
Okay, agreed.